mettle/src/stdapi/audio/apple_mic.m

```diff
@@ -0,0 +1,267 @@
+#import <AVFoundation/AVFoundation.h>
+
+#include "tlv.h"
+#include "mic.h"
+#include "log.h"
+#include "ringbuf.h"
+
+#define TLV_TYPE_AUDIO_DURATION        (TLV_META_TYPE_UINT   | TLV_EXTENSIONS + 10)
+#define TLV_TYPE_AUDIO_DATA            (TLV_META_TYPE_RAW    | TLV_EXTENSIONS + 11)
+#define TLV_TYPE_AUDIO_INTERFACE_ID    (TLV_META_TYPE_UINT   | TLV_EXTENSIONS + 12)
+#define TLV_TYPE_AUDIO_INTERFACE_NAME  (TLV_META_TYPE_STRING | TLV_EXTENSIONS + 13)
+
+#define AUDIO_MAX_BUFFER_SIZE         65536
+#define AUDIO_TARGET_SAMPLE_RATE_HZ   11025
+#define AUDIO_TARGET_BITS_PER_CHANNEL 16
+#define AUDIO_TARGET_NUM_CHANNELS     1
+
+@interface AudioCapture : NSObject <AVCaptureAudioDataOutputSampleBufferDelegate>
+- (void) captureOutput: (AVCaptureOutput*) output
+ didOutputSampleBuffer: (CMSampleBufferRef) buffer
+        fromConnection: (AVCaptureConnection*) connection;
+@end
+@interface AudioCapture ()
+{
+    AVCaptureSession* session;
```

Comment:
Annoyingly, for some reason, you can't do this on i386:
```
make TARGET=i386-apple-darwin
Building mettle for i386-apple-darwin
~/dev/git/mettle/mettle/src/stdapi/audio/apple_mic.m:25:23: error: instance variables may not be placed in class extension
    AVCaptureSession* session;
                      ^
```
I had to workaround it with this change:
https://github.com/rapid7/mettle/commit/d1f2a979044abd5ff72234e8483a27178e9dbc55#diff-c6d902dfa046d5a9253967394079fe86L24
My objective-C is a little sloppy so there might be a better fix.